test/index.js

```diff
@@ -177,19 +189,19 @@ describe('Consumer', () => {
         code: 'CredentialsError',
         message: 'Missing credentials in config'
       };
+      const timings = [];
+      const listener = sandbox.stub().callsFake(() => timings.push(new Date()));
 
-      sqs.receiveMessage.yields(credentialsErr);
+      sqs.receiveMessage = awsReject(credentialsErr);
 
-      consumer.on('error', () => {
-        setTimeout(() => {
```

Comment:
removed as many `setTimeout` calls as possible because they screw up the error reporting in mocha and create flakey tests.